### ISSUE TITLE
Repo.list doesn't show all repositories from user

### ISSUE THREAD

**Issue by joaostein** [node-gh/gh#343]
**TL;DR**: `gh re --list --user <username>` limits the response to 30 repositories, meaning that in cases when a user has more repositories than that, we won't show all of them.

When we run `gh re --list --user <username>` it limits the response to 30 repositories.  Is this the expected behavior? It seems that our GitHub wrapper methods `.getAll()` and `.getFromUser()` accepts a `payload` object with a `per_limit` key that sets the response page size (default is 30 and max 100). It also accepts a `page` key that is responsible for fetching the appropriate page, in case the user has more repositories than the `per_limit` value [[1]](https://github.com/mikedeboer/node-github/blob/master/api/v3.0.0/repos.js#L33-L34).  

In our `.list()` method [[2]](https://github.com/node-gh/gh/blob/master/lib%2Fcmds%2Frepo.js#L218-L243) we don't take in consideration the `per_limit` neither the `page` (in cases the user has more repositories than the limit).

There is a proof of concept [here](https://github.com/joaostein/gh/commit/a862a9725e05f87277f5c32ab69e740911a6792b) and if this is something we would like to pursuit, let me know and I'll issue a PR for this :)

**Comment by tomzx** [comment]
I think we should definitely be able to get all the paginated results, otherwise the tool is semi-useful.

However, I think it would be great if the fetchRepoData was done in parallel over a certain amount of pages in order to reduce the amount of time spent on the back and forth.

**Comment by wkoszek** [comment]
@joaostein Would be great to get this commited. Otherwise the CLI isn't very useful.

**Comment by henvic** [comment]
Hey guys,

I am sorry I forgot about this issue. I am going to take a look at it soon.

By the way, currently the per_page settings has been raised to 1000 (that should be good for most people).

Do you think it should show everything at once or try to paginate, say, with per_page set to a lower number such as 100 and a --page parameter?

Reference: [gh pr -a doesn't work well #413](https://github.com/node-gh/gh/issues/413)

**Comment by wkoszek** [comment]
Showing everything at once would be the best. Otherwise it's hard to script this stuff.

**Comment by henvic** [comment]
[Do one thing, and do it well](https://en.wikipedia.org/wiki/Unix_philosophy)

I just remembered about one of the best chapters I have read in [The Art of Unix Programming](http://www.catb.org/esr/writings/taoup/html/index.html) :)

The thing is, getting these results involve an expensive remote call and is not always possible.

For now I am closing this issue after @wkoszek answer. If anyone disagrees with it, feel free to reopen it and explain your motives so I can reconsider the topic.
